### PR TITLE
ZENKO-2837 Create folder

### PR DESCRIPTION
**Pull request by nicolas2bert**
Screen-Recording-2020-09-29-at-6 49 45-PM

### CODE REVIEW

**Comment by naren-rajendran** [comment]
@nicolas2bert what does `an object is created` mean here? this shows up in the popup/modal when you click create `create folder`

**Comment by nicolas2bert** [comment]
@naren-scality sorry but I don't see where I wrote `an object is created`.
As I added in the "info" section:
> When you create a folder, Data Browser creates an object with
              the above name appended by suffix &quot;/&quot; and that object is displayed
              as a folder in the Data Browser.

So, a folder is just an empty object with a "trailing slash" added to its name.

**Review by ilkescality — APPROVED** [review]
LGTM (seems it doesn't have tests but I'm guessing they'll come afterward)

File: src/react/databrowser/objects/ObjectList.jsx [hunk 1]
```diff
@@ -77,9 +77,10 @@ export default function ObjectList({ objects, bucketNameParam, prefixParam }: Pr
     }, useFilters, useSortBy);
 
     return <L.ListSection>
-        <T.SearchContainer>
-            <T.ExtraButton icon={<i className="fas fa-plus" />} text="Upload" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
-        </T.SearchContainer>
+        <T.ButtonContainer>
+            <T.ExtraButton icon={<i className="fas fa-upload" />} text="Upload" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
+            <T.ExtraButton icon={<i className="fas fa-plus" />} text="Create folder" variant='info' onClick={() => dispatch(openFolderCreateModal())} size="default" type="submit" />
```
Comment (terrydervaux) [inline src/react/databrowser/objects/ObjectList.jsx]:
```suggestion
            <T.ExtraButton icon={<i className="fas fa-plus" />} text="Create Folder" variant='info' onClick={() => dispatch(openFolderCreateModal())} size="default" type="submit" />
```

File: src/react/databrowser/objects/ObjectList.jsx [hunk 2]
```diff
@@ -77,9 +77,10 @@ export default function ObjectList({ objects, bucketNameParam, prefixParam }: Pr
     }, useFilters, useSortBy);
 
     return <L.ListSection>
-        <T.SearchContainer>
-            <T.ExtraButton icon={<i className="fas fa-plus" />} text="Upload" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
-        </T.SearchContainer>
+        <T.ButtonContainer>
+            <T.ExtraButton icon={<i className="fas fa-upload" />} text="Upload" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
```
Comment (terrydervaux) [inline src/react/databrowser/objects/ObjectList.jsx:81]:
```suggestion
            <T.ExtraButton icon={<i className="fas fa-upload" />} text="Upload Object" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
```

**Review by terrydervaux — APPROVED** [review]
LGTM except for the name of the buttons.

File: src/react/databrowser/objects/ObjectList.jsx (hunk 1 above)
Comment (nicolas2bert) [reply src/react/databrowser/objects/ObjectList.jsx]: AWS uses `Create folder` but for consistency, I agree with you and will change it to `Create Folder`.

File: src/react/databrowser/objects/ObjectList.jsx (hunk 2 above)
Comment (nicolas2bert) [reply src/react/databrowser/objects/ObjectList.jsx:81]: We have already discussed this. It is already in the second iteration ticket. https://scality.atlassian.net/browse/ZENKO-2847

**Comment by nicolas2bert** [comment]
@bert-e approve

**Pull request merged**